### ISSUE TITLE
How to implement a multi-tasks layer？

### ISSUE THREAD

**Issue by meanmee** [fchollet/keras#1532]
I have  serached the old issues，but did not get the idea.
I tried to implement it like this(two tasks for individual label):

``` python
def multi_tasks_block(input_shape):
    from keras.models import Graph
    g = Graph()
    g.add_input('input',input_shape)
    g.add_node(Dense(5, activation='relu'), 'dense','input')
    g.add_node(Dense(5, activation='softmax'),'output_dense1','dense')
    g.add_node(Dense(5, activation='softmax'),'output_dense2','dense')
    g.add_output('output1','output_dense1')
    g.add_output( 'output2','output_dense2')
    return g
nb_classes = 5
nb_epoch = 100

X_train, y_train_one, y_train_two = generate_data()
print('X_train shape:', X_train.shape)
print(X_train.shape[0], 'train samples')

Y_train_one = np_utils.to_categorical(y_train_one, nb_classes)
Y_train_two = np_utils.to_categorical(y_train_two, nb_classes)

model = Sequential()
model.add(Dense(5, input_shape=(10,), activation='relu'))
model.add(multi_tasks_block((5,)))
model.summary()

model.compile(loss = {'output1': 'categorical_crossentropy', 'ouput2': 'categorical_crossentropy'}, optimizer='adam')

print('Not using data augmentation or normalization')
model.fit(X_train, Y_train_one, batch_size=X_train.shape[0], nb_epoch=nb_epoch)
```

but I failed

**Comment by lemuriandezapada** [comment]
how did it fail? What was the error?

You're calling the fit function wrong for graph models. it's more like fit({'input':X_train', 'output1':Y_train_one, 'output2':Y_train_two},blabla)

**Comment by bienbinod** [comment]
@meanmee Did you make some progress? I am also trying to do similar stuff, but failing.

**Comment by meanmee** [comment]
```
from __future__ import print_function
from keras.datasets import cifar10
from keras.preprocessing.image import ImageDataGenerator
from keras.models import Sequential, Graph
from keras.layers.core import Dense, Dropout, Activation, Flatten, MaxoutDense
from keras.layers.convolutional import Convolution2D, MaxPooling2D, AveragePooling2D
from keras.optimizers import SGD, Adadelta, Adagrad, adam,Adam
from keras.utils import np_utils, generic_utils
import numpy as np
import matplotlib.pyplot as plt
from keras.layers.normalization import BatchNormalization
from data_utils import *
nb_classes = 5
nb_epoch = 100

X_train, y_train_one, y_train_two = generate_data()
print('X_train shape:', X_train.shape)
print(X_train.shape[0], 'train samples')

Y_train_one = np_utils.to_categorical(y_train_one, nb_classes)
Y_train_two = np_utils.to_categorical(y_train_two, nb_classes)

dense = Sequential()
dense.add(Dense(5, input_shape=(10,), activation='relu'))

model = Graph()
model.add_input('input',input_shape=(5,))
model.add_node(dense, 'dense','input')
model.add_node(Dense(5, activation='softmax'),'output_dense1','dense')
model.add_node(Dense(5, activation='softmax'),'output_dense2','dense')
model.add_output('output1','output_dense1')
model.add_output( 'output2','output_dense2')
model.summary()

# let's train the model using SGD + momentum (how original).
model.compile(optimizer=Adam(0.001),loss={'output1': 'categorical_crossentropy', 'output2': 'categorical_crossentropy'})

print('Not using data augmentation or normalization')
model.fit({'input':X_train, 'output1':Y_train_one, 'output2':Y_train_two}, batch_size=X_train.shape[0], nb_epoch=nb_epoch)
```

**Comment by bienbinod** [comment]
@meanmee Thanks for sharing your experience !
